cmd/snap-confine/mount-support.c

```diff
@@ -177,27 +177,36 @@ static void sc_setup_mount_profiles(const char *snap_name)
 {
 	debug("%s: %s", __FUNCTION__, snap_name);
 
-	FILE *f __attribute__ ((cleanup(sc_cleanup_endmntent))) = NULL;
-	const char *mount_profile_dir = "/var/lib/snapd/mount";
-
+	FILE *desired __attribute__ ((cleanup(sc_cleanup_endmntent))) = NULL;
+	FILE *current __attribute__ ((cleanup(sc_cleanup_endmntent))) = NULL;
 	char profile_path[PATH_MAX];
-	sc_must_snprintf(profile_path, sizeof(profile_path), "%s/snap.%s.fstab",
-			 mount_profile_dir, snap_name);
-
-	debug("opening mount profile %s", profile_path);
-	f = setmntent(profile_path, "r");
-	// it is ok for the file to not exist
-	if (f == NULL && errno == ENOENT) {
-		debug("mount profile %s doesn't exist, ignoring", profile_path);
+
+	sc_must_snprintf(profile_path, sizeof(profile_path),
+			 "/run/snapd/ns/snap.%s.fstab", snap_name);
+	debug("opening current mount profile %s", profile_path);
+	current = setmntent(profile_path, "w");
+	if (current == NULL) {
+		die("cannot open current mount profile: %s", profile_path);
+	}
+
+	sc_must_snprintf(profile_path, sizeof(profile_path),
+			 "/var/lib/snapd/mount/snap.%s.fstab", snap_name);
+	debug("opening desired mount profile %s", profile_path);
+	desired = setmntent(profile_path, "r");
+	if (desired == NULL && errno == ENOENT) {
+		// It is ok for the desired profile to not exist. Note that in this
+		// case we also "update" the current profile as we already opened and
+		// truncated it above.
+		debug("desired mount profile %s doesn't exist, ignoring",
```

Comment:
This message disagrees with the comment right above it. It makes it sound like nothing is happening because the file doesn't exist, while the comment says that the lack of the file is taken as an empty one and updates the current one accordingly.